src/common/data/stores/accounts/indentityStore.ts

```diff
@@ -276,6 +315,138 @@ export const indentityStore = (
+    if (isUndefined(keyPair)) {
+      const preKeys = await get().loadPreKeys();
+      keyPair = find(preKeys, { publicKey: encryptingKey });
+    }
+    // Error if key is still not found
+    if (isUndefined(keyPair)) {
+      throw new NoPreKeyFoundError(encryptingKey);
+    }
+    const cipher = managedNonce(xchacha20poly1305)(
+      stringToCipherKey(
+        bytesToHex(
+          secp256k1.getSharedSecret(keyPair.privateKey, keyPair.publicKey),
+        ),
+      ),
+    );
+    return bytesToUtf8(cipher.decrypt(utf8ToBytes(file.fileData)));
+  },
+  generatePreKey: async () => {
+    const privateKey = secp256k1.utils.randomPrivateKey();
+    const publicKey = secp256k1.getPublicKey(privateKey);
+    const prekey: PreSpaceKeys = {
+      publicKey: bytesToHex(publicKey),
+      privateKey: bytesToHex(privateKey),
+      type: "pre",
+      timestamp: moment().toISOString(),
+    };
+    const keyFile = await get().createEncryptedSignedFile(
+      stringify(prekey),
+      "json",
+      true,
+    );
+    const postData: PreKeyRequest = {
+      file: keyFile,
+      identityPublicKey: get().currentSpaceIdentityPublicKey,
+      prekeyPublicKey: bytesToHex(publicKey),
+    };
+
+    // TO DO: Error handling
+    await axiosBackend.post("/api/space/prekeys/", postData, {
+      headers: { "Content-Type": "application/json" },
+    });
+
+    get().addPreKeysToIdentity([prekey]);
+    return prekey;
+  },
+  addPreKeysToIdentity: (prekeys) => {
+    set((draft) => {
+      const currentIdentityIndex = findIndex(get().spaceIdentities, {
+        rootKeys: { publicKey: get().currentSpaceIdentityPublicKey },
+      });
+      draft.spaceIdentities[currentIdentityIndex].preKeys = sortBy(
+        concat(draft.spaceIdentities[currentIdentityIndex].preKeys, prekeys),
+        ["timestamp"],
+      );
+    });
+  },
+  loadPreKeys: async () => {
+    const keyFileLocs = await axiosBackend.get<PreKeyResponse>(
+      "/api/space/prekeys",
+      { params: { publicKey: get().getCurrentIdentity()?.rootKeys.publicKey } },
+    );
+    const supabase = createClient();
+    const prekeys = compact(
+      await Promise.all(
+        map(keyFileLocs, async (loc) => {
+          try {
+            const {
+              data: { publicUrl },
+            } = await supabase.storage.from("private").getPublicUrl(loc);
+            const { data }: { data: Blob } = await axios.get(publicUrl, {
+              responseType: "blob",
+            });
+            const fileData = JSON.parse(await data.text()) as SignedFile;
+            const decryptedFile =
+              await get().decryptEncryptedSignedFile(fileData);
+            const preKeys = JSON.parse(decryptedFile) as PreSpaceKeys;
+            return preKeys;
+          } catch (e) {
+            console.error(e);
+            return undefined;
+          }
+        }),
+      ),
+    );
+    get().addPreKeysToIdentity(prekeys);
+    return prekeys;
+  },
+  getCurrentPrekey: () => {
```

Comment:
IdentityStore is getting pretty long. Are there any natural break points that would subdivide well into smaller files?